### PR TITLE
Sub/cx 1176 router redundancy reduced

### PR DESCRIPTION
**Pull request by rfreitas**
made the code a bit dryier

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-218-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-218-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Router/index.js [hunk 1]
```diff
@@ -110,32 +97,89 @@ class MainRouter extends Component {
...
+  }
+
+  componentWillUnmount () {
+    this.unlisten()
+  }
+
+  changeFlowTo = (newFlow) => {
+    const {flow: previousFlow, step: previousStep} = this.state
+    if (previousFlow === newFlow) return
+    this.props.onFlowChange(newFlow, previousStep, previousFlow)
```
Comment (stefaniacardenas) [inline src/components/Router/index.js]: Why are you passing `previousFlow` here?

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-218-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Router/index.js [hunk 2]
```diff
@@ -2,11 +2,14 @@ import { h, Component } from 'preact'
 import { bindActionCreators } from 'redux'
 import { connect } from 'react-redux'
 import io from 'socket.io-client'
-
 import { componentsList } from './StepComponentMap'
 import { unboundActions } from '../../core'
 import StepsRouter from './StepsRouter'
 import { isDesktop } from '../utils'
+import createHistory from 'history/createBrowserHistory'
```
Comment (stefaniacardenas) [inline src/components/Router/index.js]: Minor: We are trying to keep external dependencies at the top of the file, with a line between externals and  internal ones. Could you move it after `import io from 'socket.io-client'`?

File: src/components/Router/index.js (hunk 1 above)
Comment (rfreitas) [reply src/components/Router/index.js]: not really needed to be honest, but usually event callbacks that have to purpose to remind the client a value has changed usually include the previous value as well. but it's not necessary

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-218-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-218-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-218-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**